app/backup.go

```diff
@@ -83,7 +86,21 @@ func createBackup(c *cli.Context) error {
 		}
 	}
 
-	backup, err := task.CreateBackup(snapshot, dest, labels)
+	accessKey := ""
+	secretKey := ""
+	backupType, err := checkBackupType(dest)
+	if backupType == "s3" {
+		accessKey = os.Getenv(types.AccessKey)
+		if accessKey == "" {
+			return fmt.Errorf("Missing environment variable AWS_ACCESS_KEY_ID for s3 backup")
+		}
+		secretKey = os.Getenv(types.SecretKey)
+		if secretKey == "" {
+			return fmt.Errorf("Missing environment variable AWS_SECRET_ACCESS_KEY for s3 backup")
+		}
+	}
+
+	backup, err := task.CreateBackup(snapshot, dest, labels, accessKey, secretKey)
```

Comment:
Instead of passing two variables (and potentially more in the future), we can pass a `credential map[string]string`, and the callee can perform according to which kind of protocol is it.

e.g. 
```
credential = map[string]string{}
if s3 {
  credential[types.AWSAccessKey] = xxx
  credential[types.AWSSecretKey] = xxx
}
```

Same for the `Process` struct below.